### PR TITLE
Basic maven fixes

### PR DESCRIPTION
**Pull request by ScarySpider**
Now outputs .jar files to /target:
-Server jar outputs to /target/
-All other jars output to /target/plugins

fixes simon987/Much-Assembly-Required#24

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                </configuration>
+            </plugin>
+                <plugin>
+                    <version>3.6.2</version>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <configuration>
+                        <source>1.8</source>
+                        <target>1.8</target>
+                    </configuration>
```
Comment (croog24) [inline pom.xml]: Duplicate?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+plugins/Plugin NPC.jar
+plugins/Plugin Misc HW.jar
+plugins/Plant.jar
+plugins/Cubot.jar
```
Comment (croog24) [inline .gitignore]: The below `plugins/*.jar' should suffice.

File: Server/pom.xml [hunk 3]
```diff
@@ -3,24 +3,38 @@
...
     <build>
         <plugins>
             <plugin>
+                <version>3.6.2</version>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
                     <source>1.8</source>
                     <target>1.8</target>
                 </configuration>
```
Comment (croog24) [inline Server/pom.xml]: This compatibility plugin can be removed like the others since it is inherited.

**Comment by simon987** [comment]
Thanks a lot for the help with the maven part!

**Comment by ScarySpider** [comment]
No problem. Let me know if/when you need anything else done to Maven.

**Pull request merged**